packages/txservice/src/transaction.ts

```diff
@@ -0,0 +1,216 @@
+import { delay, jsonifyError } from "@connext/nxtp-utils";
+import { BigNumber, providers } from "ethers";
+import { BaseLogger } from "pino";
+
+import { TransactionServiceConfig } from "./config";
+import { ChainError } from "./error";
+import { ChainRpcProvider } from "./provider";
+import { FullTransaction, MinimalTransaction } from "./types";
+
+export class Transaction {
+
+  public responses: providers.TransactionResponse[] = [];
+  public receipt?: providers.TransactionReceipt;
+  public nonce?: number;
+  public nonceExpired = false;
+
+  // TODO: Not a fan of this arrangement, this code could be tightened up a bit.
+  public get gasPrice(): Promise<BigNumber> {
+    return this.getGasPrice();
+  }
+  private _gasPrice?: BigNumber;
+
+  public get fullTransaction(): Promise<FullTransaction> {
```

Comment:
Is it weird for a getter like this to be async? Should it just be a private method?

Reply:
I was thinking it'd be useful for external calling in the event we want to log this transaction's properties